mk/config.mk

```diff
@@ -369,6 +369,19 @@ CFG_TA_GPROF_SUPPORT ?= n
 # defined in tee-supplicant)
 CFG_FTRACE_SUPPORT ?= n
 
+# Size of the function tracing buffer in bytes. The memory size required to
+# map the TA is increased by this value compared to when ftrace is disabled.
+CFG_FTRACE_BUF_SIZE ?= 2048
+
+# How to make room when the function tracing buffer is full?
+# 'shift': shift the previously stored data by the amount needed in order
+#    to always keep the latest logs (slower, especially with big buffer sizes)
+# 'wrap': discard the previous data and start at the beginning of the buffer
+#    again (fast, but can result in a mostly empty buffer)
+# 'stop': stop logging new data
+CFG_FTRACE_BUF_WHEN_FULL ?= shift
+$(call force,CFG_FTRACE_BUF_WHEN_FULL_$(CFG_FTRACE_BUF_WHEN_FULL),y)
```

Comment:
We should have a test somewhere that only one of `CFG_FTRACE_BUF_WHEN_FULL_{shift, wrap, stop}` is used

Reply:
There is https://github.com/OP-TEE/optee_os/pull/3601/files#diff-5fe3f587610e26f68128dd3201a33397R105

Do you mean you'd rather have it checked in `mk/config.mk`? (I would tend to agree...)

Reply:
Yes, I'd prefer to have the test in a makefile. If it's short enough I suppose we can have it here else perhaps `core/core.mk`

Reply:
How about checking that:
`$(filter y,$(CFG_FTRACE_BUF_WHEN_FULL_shift) $(CFG_FTRACE_BUF_WHEN_FULL_stop) $(CFG_FTRACE_BUF_WHEN_FULL_stop))` == y

Reply:
That would somewhat redundant with what I have added (`cfg-check-value`), which also provides a good error message.
Ah, maybe I should prefix those derived variables with an underscore, so they wouldn't even be visible in the generated `$(out-dir)/conf.mk` ? (we do have a few already).

Reply:
Sounds good